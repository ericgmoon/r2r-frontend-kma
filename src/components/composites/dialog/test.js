import React from "react";
import sinon from "sinon";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import Dialog from "./index";

describe("should trigger onDialogFinish", () => {
  const callback = sinon.spy();

  test("with one screen", () => {
    const node = mount(
      <Dialog
        content={[
          {
            name: "Person 1",
            text: "Lorem ipsum dolor sit amet",
          },
        ]}
        className="dialog"
        onDialogFinish={callback}
      />
    );
    expect(node.find(".dialog")).toBeDefined();
    node
      .find(".dialog")
      .findWhere((n) => n.prop("testID") === "nextButton")
      .first()
      .props()
      .onPress();
    sinon.assert.called(callback);
  });

  test("with multiple screens", () => {
    const node = mount(
      <Dialog
        content={[
          {
            name: "Person 1",
            text: "Lorem ipsum dolor sit amet",
          },
          {
            name: "Person 2",
            text: "Lorem ipsum dolor sit amet",
          },
        ]}
        className="dialog"
        onDialogFinish={callback}
      />
    );
    expect(node.find(".dialog")).toBeDefined();
    // Navigate to the next page
    act(async () => {
      await node
        .find(".dialog")
        .findWhere((n) => n.prop("testID") === "nextButton")
        .first()
        .props()
        .onPress();
      await sinon.assert.notCalled(callback);
    });
    // Press the end button
    node
      .find(".dialog")
      .findWhere((n) => n.prop("testID") === "nextButton")
      .first()
      .props()
      .onPress();
    sinon.assert.called(callback);
  });

  describe("renders correctly", () => {
    test("without optional props", () => {
      const node = renderer
        .create(
          <Dialog
            content={[
              {
                name: "Person 1",
                text: "Lorem ipsum dolor sit amet",
              },
              {
                name: "Person 2",
                text: "Lorem ipsum dolor sit amet",
              },
            ]}
            onDialogFinish={() => {}}
          />
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with long texts", () => {
      const node = renderer
        .create(
          <Dialog
            content={[
              {
                name: "Person 1",
                text: "Lorem ipsum dolor sit amet",
              },
              {
                name: "Person 2",
                text:
                  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
              },
              {
                name: "Person 1",
                text: "Lorem ipsum dolor sit amet",
              },
            ]}
            onDialogFinish={() => {}}
          />
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with custom name colors", () => {
      const node = renderer
        .create(
          <Dialog
            content={[
              {
                name: "Person 1",
                nameColor: "#6abd75",
                text: "Lorem ipsum dolor sit amet",
              },
              {
                name: "Person 2",
                nameColor: "#7274b5",
                text: "Lorem ipsum dolor sit amet",
              },
            ]}
            onDialogFinish={() => {}}
          />
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with avatars", () => {
      const node = renderer
        .create(
          <Dialog
            content={[
              {
                name: "Person 1",
                avatar: "sample",
                text: "Lorem ipsum dolor sit amet",
              },
              {
                name: "Person 2",
                avatar: "sample",
                isLeft: true,
                text: "Lorem ipsum dolor sit amet",
              },
              {
                name: "Person 1",
                avatar: "sample",
                text: "Lorem ipsum dolor sit amet",
              },
            ]}
            avatars={{
              // eslint-disable-next-line global-require, import/no-unresolved
              sample: require("@r2r/assets/images/dev/sample-dialog-avatar.png"),
            }}
            onDialogFinish={() => {}}
          />
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with custom buttons", () => {
      const node = renderer
        .create(
          <Dialog
            content={[
              {
                name: "Person 1",
                text: "Lorem ipsum dolor sit amet",
                buttonText: "Continue",
              },
              {
                name: "Person 2",
                text: "Lorem ipsum dolor sit amet",
                buttonText: "Start",
                buttonIcon: "play-arrow",
              },
            ]}
            onDialogFinish={() => {}}
          />
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });
  });
});
