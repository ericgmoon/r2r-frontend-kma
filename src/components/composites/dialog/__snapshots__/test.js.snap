// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should trigger onDialogFinish renders correctly with avatars 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "40%",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DF6A6A",
              "borderColor": "#2089dc",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 8,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 8,
              "textTransform": "uppercase",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Person 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "marginTop": -16,
        "overflow": "hidden",
        "padding": 32,
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={3}
      onTextLayout={[Function]}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 12,
          "marginTop": 12,
          "minHeight": "30%",
        }
      }
    >
      Lorem ipsum dolor sit amet
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/dev/sample-dialog-avatar.png",
        }
      }
      style={
        Array [
          Object {
            "height": "100%",
            "width": 96,
          },
          Object {
            "alignSelf": "flex-end",
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="nextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "background": "#000",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            NEXT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should trigger onDialogFinish renders correctly with custom buttons 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "40%",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DF6A6A",
              "borderColor": "#2089dc",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 8,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 8,
              "textTransform": "uppercase",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Person 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "marginTop": -16,
        "overflow": "hidden",
        "padding": 32,
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={3}
      onTextLayout={[Function]}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 12,
          "marginTop": 12,
          "minHeight": "30%",
        }
      }
    >
      Lorem ipsum dolor sit amet
    </Text>
    <Image
      resizeMode="contain"
      style={
        Array [
          Object {
            "height": "100%",
            "width": 96,
          },
          Object {
            "alignSelf": "flex-end",
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="nextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "background": "#000",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            CONTINUE
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should trigger onDialogFinish renders correctly with custom name colors 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "40%",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#6abd75",
              "borderColor": "#2089dc",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 8,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 8,
              "textTransform": "uppercase",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Person 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "marginTop": -16,
        "overflow": "hidden",
        "padding": 32,
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={3}
      onTextLayout={[Function]}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 12,
          "marginTop": 12,
          "minHeight": "30%",
        }
      }
    >
      Lorem ipsum dolor sit amet
    </Text>
    <Image
      resizeMode="contain"
      style={
        Array [
          Object {
            "height": "100%",
            "width": 96,
          },
          Object {
            "alignSelf": "flex-end",
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="nextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "background": "#000",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            NEXT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should trigger onDialogFinish renders correctly with long texts 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "40%",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DF6A6A",
              "borderColor": "#2089dc",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 8,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 8,
              "textTransform": "uppercase",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Person 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "marginTop": -16,
        "overflow": "hidden",
        "padding": 32,
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={3}
      onTextLayout={[Function]}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 12,
          "marginTop": 12,
          "minHeight": "30%",
        }
      }
    >
      Lorem ipsum dolor sit amet
    </Text>
    <Image
      resizeMode="contain"
      style={
        Array [
          Object {
            "height": "100%",
            "width": 96,
          },
          Object {
            "alignSelf": "flex-end",
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="nextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "background": "#000",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            NEXT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should trigger onDialogFinish renders correctly without optional props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "40%",
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DF6A6A",
              "borderColor": "#2089dc",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 8,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 8,
              "textTransform": "uppercase",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Person 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": "100%",
        "marginTop": -16,
        "overflow": "hidden",
        "padding": 32,
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={3}
      onTextLayout={[Function]}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 12,
          "marginTop": 12,
          "minHeight": "30%",
        }
      }
    >
      Lorem ipsum dolor sit amet
    </Text>
    <Image
      resizeMode="contain"
      style={
        Array [
          Object {
            "height": "100%",
            "width": 96,
          },
          Object {
            "alignSelf": "flex-end",
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 36,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="nextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "background": "#000",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 36,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingBottom": 16,
              "paddingLeft": 20,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            NEXT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
